physics/inertia_tensor.hpp

```diff
@@ -0,0 +1,111 @@
+﻿#pragma once
+
+#include "geometry/named_quantities.hpp"
+#include "geometry/point.hpp"
+#include "geometry/r3_element.hpp"
+#include "geometry/r3x3_matrix.hpp"
+#include "geometry/rotation.hpp"
+#include "geometry/symmetric_bilinear_form.hpp"
+#include "quantities/named_quantities.hpp"
+
+namespace principia {
+namespace physics {
+namespace internal_inertia_tensor {
+
+using geometry::Displacement;
+using geometry::Position;
+using geometry::R3Element;
+using geometry::R3x3Matrix;
+using geometry::Rotation;
+using geometry::SymmetricBilinearForm;
+using quantities::Mass;
+using quantities::MomentOfInertia;
+
+// The inertia tensor of a solid at the origin and in the axis of Frame.
+template<typename Frame>
+class InertiaTensor {
+ public:
+  // Constructs an inertia tensor with the specified coordinates for an object
+  // of the given mass.  The centre of mass of the object must be specified.
+  // Typically, this constructor is used to implement formulæ like those in
+  // https://en.wikipedia.org/wiki/List_of_moments_of_inertia and the centre of
+  // mass is often Frame::origin.
+  InertiaTensor(Mass const& mass,
+                R3x3Matrix<MomentOfInertia> const& coordinates,
+                Position<Frame> const& centre_of_mass);
+
+  // Obtains the centre of mass of the object in Frame.
+  Position<Frame> const& centre_of_mass() const;
+
+  // Computes the inertia tensor in ToFrame, which is rotated from Frame using
+  // the given rotation, with the origins of the frames coïnciding.  This
+  // effectively defines ToFrame.
+  template<typename ToFrame>
+  InertiaTensor<ToFrame> Rotate(Rotation<Frame, ToFrame> const& rotation) const;
+  template<typename ToFrame>
+  InertiaTensor<ToFrame> Rotate(Rotation<ToFrame, Frame> const& rotation) const;
+
+  // Computes the inertia tensor in ToFrame, which is translated from Frame
+  // such that its origin is at the given points, with the axes of the frames
+  // coïnciding.  This effectively defines ToFrame.
+  template<typename ToFrame>
+  InertiaTensor<ToFrame> Translate(Position<Frame> const& point) const;
```

Comment:
Expose a function that transforms the inertia tensor with a given `RigidTransformation`, keeping `Translate` and `Rotate` as underlying internals.

Reply:
Done (well, `Translate` and `Rotate` are gone).